joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h

```diff
@@ -773,6 +775,15 @@ updateStates(const ros::Time& sample_time, const Trajectory* const traj)
     state_error_.position[joint_index] = angles::shortest_angular_distance(current_state_.position[joint_index],desired_joint_state_.position[0]);
     state_error_.velocity[joint_index] = desired_joint_state_.velocity[0] - current_state_.velocity[joint_index];
     state_error_.acceleration[joint_index] = 0.0;
+
+    if (joint_index == 0)
+    {
+      const auto time_from_start = segment->timeFromStart();
+      current_state_.time_from_start = sample_time.toSec() - segment->startTime() + time_from_start;
+      desired_state_.time_from_start = time_from_start;
+    }
+
+    state_error_.time_from_start = desired_state_.time_from_start - current_state_.time_from_start;
```

Comment:
I think this is better in the `if` statement above.

Reply:
good catch, added it there, removed from here!

```suggestion
```